### PR TITLE
[GoogleAnalytics] covered Helper Data by Unit Test

### PR DESCRIPTION
**Pull request by srsathish92**
### Description (*)
[GoogleAnalytics] covered Helper Data by Unit Test with data provider
Path: `app/code/Magento/GoogleAnalytics/Helper/Data.php`

### Manual testing scenarios (*)
`vendor/bin/phpunit -c dev/tests/unit/phpunit.xml.dist app/code/Magento/GoogleAnalytics/Test/Unit/Helper/DataTest.php`

### Questions or comments
If any please let me know your feedback

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/GoogleAnalytics/Test/Unit/Helper/DataTest.php [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            ->method('getValue')
+            ->with(HelperData::XML_PATH_ACCOUNT, ScopeInterface::SCOPE_STORE)
+            ->willReturn($value);
+
+        $this->scopeConfigMock->expects($this->any())
+            ->method('isSetFlag')
+            ->with(HelperData::XML_PATH_ACTIVE, ScopeInterface::SCOPE_STORE)
+            ->willReturn($flag);
+
+        $this->assertEquals(($value && $flag), $this->helper->isGoogleAnalyticsAvailable());
```
Comment (dmytro-ch) [inline app/code/Magento/GoogleAnalytics/Test/Unit/Helper/DataTest.php]: Please replace this operation `($value && $flag)` with the exat expected result value provided by `gaDataProvider `.

e.g.
```
return [
            ['GA-XXXX', true, true],
            ['GA-XXXX', false, false],
            ['', true, false]
        ];
```

File: app/code/Magento/GoogleAnalytics/Test/Unit/Helper/DataTest.php [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+     * @return void
+     * @dataProvider yesNoDataProvider
+     */
+    public function testIsAnonymizedIpActive($value): void
+    {
+        $this->scopeConfigMock->expects($this->once())
+            ->method('getValue')
+            ->with(HelperData::XML_PATH_ANONYMIZE, ScopeInterface::SCOPE_STORE)
+            ->willReturn($value);
+        $this->assertEquals((bool) $value, $this->helper->isAnonymizedIpActive());
```
Comment (dmytro-ch) [inline app/code/Magento/GoogleAnalytics/Test/Unit/Helper/DataTest.php]: Please replace this one (`(bool) $value`) with the exact expected result provided by `yesNoDataProvider`.
So the input params will be $value and $result.
```
public function testIsAnonymizedIpActive($value, $result): void

**Review by dmytro-ch — CHANGES_REQUESTED** [review]

File: app/code/Magento/GoogleAnalytics/Test/Unit/Helper/DataTest.php (hunk 1 above)
Comment (srsathish92) [reply app/code/Magento/GoogleAnalytics/Test/Unit/Helper/DataTest.php]: @dmytro-ch Thanks for the feedback. I have updated the changes. please review now

File: app/code/Magento/GoogleAnalytics/Test/Unit/Helper/DataTest.php (hunk 2 above)
Comment (srsathish92) [reply app/code/Magento/GoogleAnalytics/Test/Unit/Helper/DataTest.php]: @dmytro-ch Thanks for the feedback. I have updated the changes. please review now

**Comment by dmytro-ch** [comment]
@srsathish92, thank you for the updates!

**Review by dmytro-ch — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @dmytro-ch, thank you for the review.
[ENGCOM-6495](https://jira.corp.magento.com/browse/ENGCOM-6495) has been created to process this Pull Request

**Comment by engcom-Delta** [comment]
**Notice:** QA not applicable

**Pull request merged**